src/planscape/planning/views.py

```diff
@@ -182,3 +181,196 @@ def list_plans(request: HttpRequest) -> HttpResponse:
+                case ScenarioResultStatus.SUCCESS:
+                    if old_status != ScenarioResultStatus.RUNNING:
+                        raise ValueError("Invalid new state.")
+                case _:
+                    if new_status != ScenarioResultStatus.FAILURE:
+                        raise ValueError("Invalid new state.")
+            scenario_result.status = new_status
+            
+        if (run_details := body.get('run_details')) is not None:
+            scenario_result.run_details = run_details
+ 
+        if (result := body.get('result')) is not None:
+            scenario_result.result = result
+            
+        scenario_result.save()
+
+        return HttpResponse(str(scenario_id))
+    except Exception as e:
+        return HttpResponseBadRequest("Update Scenario error: " + str(e))
+
+
+# This returns the empty set if given a plan that has no scenarios,
+# a plan that isn't owned by the user, or a planID that doesn't exist.
+def list_scenarios_for_plan(request: HttpRequest) -> HttpResponse:
+    try:
+        # Check that the user is logged in.
+        user = _get_user(request)
+        if user is None:
+            raise ValueError("User must be logged in.")
+
+        plan_id = request.GET['plan_id']
+        if plan_id is None:
+            raise ValueError("Missing plan_id")
+
+        scenarios = Scenario.objects.filter(planning_area__user_id=user.pk).filter(planning_area__pk=plan_id)
+        
+        return JsonResponse(
+            [_serialize_scenario(scenario, None) for scenario in scenarios],
+            safe=False)
+    except Exception as e:
+        return HttpResponseBadRequest("List Scenario error: " + str(e))
+
+# Can delete multiple scenarios at once.
+# Deletion attempts for someone else's scenario will silently do nothing with those scenarios.
+def delete_scenario(request: HttpRequest) -> HttpResponse:
+    try:
+        # Check that the user is logged in.
+        user = _get_user(request)
+        if user is None:
+            raise ValueError("User must be logged in.")
+        
+        body = json.loads(request.body)
+        scenario_id_str = body.get('scenario_id', None)
+        if scenario_id_str is None:
+            raise ValueError("Must specify scenario id(s)")
+
+        scenario_ids = []
+        if isinstance(scenario_id_str, int):
+            scenario_ids = [scenario_id_str]
+        elif isinstance(scenario_ids, list):
+            scenario_ids = scenario_id_str
+        else:
+            raise ValueError("scenario_id must be an int or a list of ints.")
+
+        # Get the scenarios matching the provided IDs and the logged-in user.
+        scenarios = Scenario.objects.filter(pk__in=scenario_ids).filter(planning_area__user=user.pk)
+        # This automatically deletes ScenarioResult entries for the deleted Scenarios.
+        scenarios.delete()
+
+        # We still report that the full set of plan IDs requested were deleted,
```

Comment:
Can you clarify this comment a bit? Is it saying even if the user is wrong, it still reports the objects were deleted? 

(Also "plan ID's" -> "scenario ID's"?)

Reply:
Reworded.